sql/core/src/main/scala/org/apache/spark/sql/execution/window/WindowExecBase.scala

```diff
@@ -137,7 +137,11 @@ trait WindowExecBase extends UnaryExecNode {
           function match {
             case AggregateExpression(f, _, _, _, _) => collect("AGGREGATE", frame, e, f)
             case f: AggregateWindowFunction => collect("AGGREGATE", frame, e, f)
-            case f: OffsetWindowFunction => collect("OFFSET", frame, e, f)
+            case f: OffsetWindowFunction => if (f.isWholeBased) {
```

Comment:
are we going to add more functions like `NthValue`? If not I think it's simpler to do

```
case f: NthValue => collect("WHOLE_OFFSET", frame, e, f)
case f: OffsetWindowFunction => collect("OFFSET", frame, e, f)
```

so that we don't need to add the `isWholeBased` API.

Reply:
According to the plan, `first_value` and `last_value` need to be realized.